frontend/lib/src/components/elements/GraphVizChart/GraphVizChart.tsx

```diff
@@ -38,75 +38,68 @@ interface Dimensions {
 const dummyGraphviz = graphviz
 dummyGraphviz // eslint-disable-line @typescript-eslint/no-unused-expressions
 
+const isFullScreen = (height = 0): boolean => Boolean(height)
+
 export function GraphVizChart({
   width: propWidth,
   element,
   height: propHeight,
 }: GraphVizChartProps): ReactElement {
   const chartId = `graphviz-chart-${element.elementId}`
 
+  const isFull = isFullScreen(propHeight)
+  const isFullRef = useRef(isFull)
```

Comment:
Why `useRef`? Since you don't seem to be saving an HTMLElement reference, think it might make more sense to condense `isFull` and `isFullRef`, subbing for `useState`

Reply:
> Since you don't seem to be saving an HTMLElement reference

`useRef` is used here to hold a mutable reference to the `isFullScreen` value which does not cause a re-render when the reference is changed. We want to persist the value across renders without causing a re-render. 

Since `setSvgDimensions` depends on `isFullScreenRef`, which is mutable and will not trigger a re-render when changed, the effect that has `setSvgDimensions` as a dependency will not re-run unnecessarily when `isFullScreen` changes. 

Condensing `isFullScreen` and `isFullScreenRef`, subbing for `useState` causes buggy fullscreen behavior due to re-rendering.